src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -42,11 +175,22 @@ export default function ConfirmSwapModal({
   fiatValueInput: { data?: number; isLoading: boolean }
   fiatValueOutput: { data?: number; isLoading: boolean }
 }) {
-  const showAcceptChanges = useMemo(
-    () => Boolean(originalTrade && tradeMeaningfullyDiffers(trade, originalTrade)),
-    [originalTrade, trade]
+  const doesTradeDiffer = originalTrade && tradeMeaningfullyDiffers(trade, originalTrade, allowedSlippage)
+  const { onStartSwapFlow, onCancel, confirmModalState, approvalError, pendingModalSteps, prepareSwapFlow } =
+    useConfirmModalState({
+      trade,
+      allowedSlippage,
+      onSwap: onConfirm,
+      allowance,
+      doesTradeDiffer: Boolean(doesTradeDiffer),
+    })
+
+  const showAcceptChanges = Boolean(
+    trade && doesTradeDiffer && confirmModalState !== ConfirmModalState.PENDING_CONFIRMATION
   )
 
+  const confirmed = useIsTransactionConfirmed(txHash)
```

Comment:
Will this keep running multiple times every render? Is this desired?

Reply:
i think this only updates when the `txHash` changes or when the result of `useAllTransactions` hooks changes. i wasn't sure what the improvement should be here, but i moved this hook into the child component where it's actually used